validator/src/journal/candidate_block.rs

```diff
@@ -49,7 +49,25 @@ pub struct FinalizeBlockResult {
     pub injected_batch_ids: Vec<String>,
 }
 
-pub struct CandidateBlock {
+pub trait CandidateBlock: Send + Sync {
+    fn cancel(&mut self);
+
+    fn previous_block_id(&self) -> String;
+
+    fn add_batch(&mut self, batch: Batch);
+
+    fn can_add_batch(&self) -> bool;
+
+    fn summarize(&mut self, force: bool) -> Result<Option<Vec<u8>>, CandidateBlockError>;
+
+    fn finalize(
+        &mut self,
+        consensus_data: &[u8],
+        force: bool,
+    ) -> Result<FinalizeBlockResult, CandidateBlockError>;
+}
+
+pub struct SawtoothCandidateBlock {
```

Comment:
It is all sawooth... how about PyCandidateBlock?

Reply:
After offline discussion, chnage it to FFICandidateBlock